datanommer.models/datanommer/models/__init__.py

```diff
@@ -128,7 +131,14 @@ def add(envelope):
     obj.msg = message['msg']
     obj.headers = headers
 
-    session.add(obj)
+    try:
+        session.add(obj)
+        session.flush()
```

Comment:
My experience is similar to the one of @mikebonnet I basically use ``.flush()`` to find out database integrity error (or duplicate error or...) while still giving me the possibility to rollback the transaction later on, which a commit would not allow.